tools/crd-client-generator-js/src/generateClient.ts

```diff
@@ -1,59 +1,90 @@
-const YAML = require('yaml');
-const fs = require('fs');
-import Generator from 'openapi-to-flowtype/dist/Generator';
-const prettier = require('prettier');
+const YAML = require("yaml");
+const fs = require("fs");
+import Generator from "openapi-to-flowtype/dist/Generator";
+const prettier = require("prettier");
 
 export const DEFAULT_PRETTIER_OPTIONS = {
-  parser: 'babel',
+  parser: "babel",
 };
 
-export function generateClient(crdFile: string, destinationFile: string, prefix?: string) {
+export function generateClient(
+  crdFile: string,
+  destinationFile: string,
+  prefix?: string
+) {
   const prettierOptions = {
     ...DEFAULT_PRETTIER_OPTIONS,
-    ...( prettier.resolveConfig.sync( destinationFile ) || {} )
+    ...(prettier.resolveConfig.sync(destinationFile) || {}),
   };
-  
-  const file = fs.readFileSync(crdFile, 'utf8');
+
+  const file = fs.readFileSync(crdFile, "utf8");
   const crdSpec = YAML.parse(file);
-  
+
   const generator = new Generator();
+
+  const versions =
+    crdSpec.apiVersion === "apiextensions.k8s.io/v1"
+      ? crdSpec.spec.versions.map((version) => version.name)
+      : [crdSpec.spec.version];
+  const jsonSchemas =
+    crdSpec.apiVersion === "apiextensions.k8s.io/v1"
+      ? crdSpec.spec.versions.reduce(
+          (aggregatedJsonSchemas, version) => ({
+            ...aggregatedJsonSchemas,
+            [version.name]: version.schema.openAPIV3Schema,
+          }),
+          {}
+        )
+      : { [crdSpec.spec.version]: crdSpec.spec.validation.openAPIV3Schema };
```

Comment:
Same here: `spec.validation` may describe a top-level schema for all versions, but it's an optional field, and `v1beta1` CRDs can use `spec.versions[*].schema` instead.